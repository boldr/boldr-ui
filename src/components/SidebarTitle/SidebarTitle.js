/* @flow */
import React from 'react';
import classnames from 'classnames';

import { StyleClasses } from '../../theme/styleClasses';

type Props = {
  className: ?string,
  text: string,
};

const BASE_ELEMENT = StyleClasses.SIDEBAR_TITLE;

const SidebarTitle = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  return (
    <li className={ classes }>
      { props.text }
    </li>
  );
};

export default SidebarTitle;
