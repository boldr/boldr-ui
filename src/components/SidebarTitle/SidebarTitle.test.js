import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SidebarTitle from './SidebarTitle';

describe('<SidebarTitle />', () => {
  const wrapper = shallow(<SidebarTitle text="widget" />);
  it('renders <SidebarTitle /> without breaking', () => {
    expect(wrapper.find('li').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
