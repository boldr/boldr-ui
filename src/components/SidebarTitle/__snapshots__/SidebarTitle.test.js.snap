// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarTitle /> renders snapshot 1`] = `
<li
  className="boldrui-sidebar-title"
>
  widget
</li>
`;
